src/wallet/wallet2.cpp

```diff
@@ -162,7 +162,7 @@ struct options {
   };
 };
 
-void do_prepare_file_names(const std::string& file_path, std::string& keys_file, std::string& wallet_file)
+void do_prepare_file_names(const std::string& file_path, std::string& keys_file, std::string& wallet_file, std::string &mms_file)
```

Comment:
I don't think it's good idea to make such intrusive dependency into `wallet2` module

Reply:
Can you please elaborate on that? Is this a narrow and specific comment about this particular method and about changing its signature, or a quite general comment about the relationship between a `message_store` and a `wallet2` object?

It has been quite difficult to get high-level feedback about this, but this is at the heart of my proposal: I propose to be the MMS to be a new *integral* and *standard* part of Monero and its wallets. After the introduction of the MMS, a Monero wallet has a key file, a wallet file, and optionally a message store file, and at least as I see it, it's allowed to be "open" about that, no need to hide something.

Reply:
> Is this a narrow and specific comment about this particular method and about changing its signature, or a quite general comment about the relationship between a message_store and a wallet2 object?

I think it's better to save some MMS related data by MMS module itself and make `wallet2` know nothing about it. I read you had some issues with making the module as much independent from `wallet2` as possible (that's why you introduced new interface). I think it's worth to think about making some methods in `wallet2` to export data needed for MMS module instead of inheriting from the interface (because it's not `kind-of` relation). I haven't had a chance to dive in the architecture at the moment. Probably later I'll propose some more valuable suggestions.

Reply:
>  I propose to be the MMS to be a new integral and standard part of Monero and its wallets.

It may be standard part of monero, but it's better to keep it separate from `wallet2` class because, in my opinion, `wallet2` has a lot of responsibilities already and it's not easy to test and keep free of bugs.

Reply:
The architecture currently is like that:

The functionality added to `wallet2` is quite simple. The most important new job of a `wallet2` object is managing a new third file beside the key file and the wallet file: the message store file. A `wallet2` now "owns" a message store (file).

Which is, IMHO, useful: Because `wallet2` manages the file all the programs that may use the MMS don't have to, no danger that the message store file "gets lost" or does not get renamed when a wallet is renamed etc.

Why should it have advantages if all wallet programs using `wallet2` (I know about 5 such programs) have to babysit a message store beside it themselves when `wallet2` could easily do that job once and for all its clients?

Most of the new functionality is 1) in the `message_store` class and 2) in any wallet program like in my case the CLI wallet that want to support the MMS.

If you don't use the MMS **no** message store file will be created - you won't notice anything, `wallet2` will work as it ever has, with two files. Only by initializing the MMS a message store file is created. Technically, if you delete the file, nothing bad happens, the MMS is just not active anymore for that wallet.